subsys/bluetooth/mesh/cfg_srv.c

```diff
@@ -2660,7 +2660,8 @@ static void send_friend_status(struct bt_mesh_model *model,
 	struct bt_mesh_cfg_srv *cfg = model->user_data;
 
 	bt_mesh_model_msg_init(&msg, OP_FRIEND_STATUS);
-	net_buf_simple_add_u8(&msg, cfg->frnd);
+	net_buf_simple_add_u8(&msg,
+			      cfg ? cfg->frnd : BT_MESH_FRIEND_NOT_SUPPORTED);
```

Comment:
If checking `cfg != NULL` is really necessary (in which case we should make sure it's done everywhere) shouldn't it be an assert instead of a runtime check? How can you even get to this situation considering that `cfg_srv_init()` will fail if there's no context provided, i.e. you shouldn't be able to exercise any of the cfg_srv code in this case.

Reply:
Hmm, is the issue that `mod_init()` in `access.c` doesn't propagate errors from the per-module init callback upward? I have a feeling we should fail `bt_mesh_init()` if this happens.

Reply:
The return value from the init function is ignored by access now, so even if we fail init, we'll still be able to receive mesh messages. 

I agree though, this should be handled on init and removed everywhere else. Do we need additional handling in init in your opinion?

Reply:
If any of the module init callbacks fails it should be an indication that something's not correctly configured or set up, so I'd just propagate the error all the way up to the caller of bt_mesh_init(). Not sure what exactly you mean by additional handling?

Reply:
Well, there's another option as well: forget about error propagation and make the init callbacks do asserts instead. Not sure which option is more appropriate.

Reply:
Global failure sounds right to me. The init callbacks have a return code, so there's no reason to add another way to indicate failure, I think. A third option is to disable only the failing model, but this would probably be a spec violation.

Reply:
So I can make a separate pull request implementing the suggested error propagation, but that would be a functional change. Could this PR be merged as is before 2.3.0 then, to fix the reported issue, then the proper fix with error returns can come after the code freeze is lifted?

Reply:
What bothers me is that this isn't fixing really anything. All it's doing is hiding an application bug (missing configuration server context).

Reply:
I have implemented the required changes for init handling in #25365. 
It's a functional change in the model API, so it should probably wait until after the release. 
I'll close this one.